File: pythons/deret.py
hasil = 0
strA = []

while True : 
    n = input("Masukkan jumlah deret (keluar ketik 'q') :")
    if n == 'q':
        break
    try : 
        n = int(n)
    except ValueError :
        print("Input tidak valid. Harap masukkan angka.")   
        continue
    for i in range (1, n+1):
        a = 5**i
        hasil = hasil + a
        strA.append("5^" + str(i))
    
    print("Deret yang dihitung:")
    print(" + ".join(strA) + " = ", hasil)
    print()

File: javas/deret.java
import java.util.Scanner;
import java.util.ArrayList;
import java.math.BigInteger;

public class deret {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        BigInteger hasil = BigInteger.ZERO;
        ArrayList<String> strA = new ArrayList<>();

        while (true) {
            System.out.print("Masukkan jumlah deret (keluar ketik 'q') :");
            String input = sc.nextLine();

            if (input.equals("q")) {
                break;
            }

            int n;
            try {
                n = Integer.parseInt(input);
            } catch (NumberFormatException e) {
                System.out.println("Input tidak valid. Harap masukkan angka.");
                continue;
            }

            for (int i = 1; i <= n; i++) {
                BigInteger a = BigInteger.valueOf(5).pow(i);
                hasil = hasil.add(a);
                strA.add("5^" + i);
            }

            System.out.println("Deret yang dihitung:");
            System.out.println(String.join(" + ", strA) + " = " + hasil);
            System.out.println();
        }

        sc.close();
    }
}
